util/pkg/vfs/s3fs.go

```diff
@@ -579,6 +580,13 @@ type terraformS3File struct {
 	Provider *terraformWriter.Literal `json:"provider,omitempty" cty:"provider"`
 }
 
+type terraformDOFile struct {
```

Comment:
Do you happen to know why DO reuses S3Path rather than implement its own DOPath?

This is where it is reused: https://github.com/kubernetes/kops/blob/e63cb44bd06e4b5d079f53f164d526035d29f84e/util/pkg/vfs/context.go#L348-L364

Reply:
I think it's because we could access DO Spaces using S3 complaint API and that was already in-place and we started using that. We didn't go back and change that to use DO API's instead. May be take this separately?

Reply:
yea a separate PR would be fine 👍🏻